example/main.go

```diff
@@ -12,41 +11,40 @@ import (
 
 func main() {
 	vecty.RenderBody(&body{
-		store: vstore.CreateStore(&store.State{}),
+		store: vstore.New(&store.State{}),
 	})
 }
 
 type body struct {
 	vecty.Core
-	store *vstore.Store
+	store vstore.Store
 }
 
 func (b *body) Render() *vecty.HTML {
 	return elem.Body(
-		b.store.NewComponent(NewMainComp),
+		b.store.Connect(NewMainComp()),
 	)
 }
 
 type mainComp struct {
 	vecty.Core
-	Number   int `vecty:"prop"`
-	Dispatch func(action interface{})
+	store vstore.Store
 }
 
 // NewMainComp is the mainComp constructor
-func NewMainComp(s *vstore.Store) vecty.Component {
-	state := s.GetState().(*store.State)
+func NewMainComp() vstore.StoreComponent {
```

Comment:
this is not needed anymore. we can just pass `&mainComp{}` in the `body`'s `Render` method. We should also not pass the entire `store` anymore since we only use the `Dispatch` method, so we can just pass `Dispatch` to be explicit about what this component needs.

Reply:
Yes, it just demonstrates the user can have addition logic in the initialize method.